### ISSUE TITLE
signalrcore client doesn't work with Python2.7

### ISSUE THREAD

**Issue by gvanderpol** [mandrewcito/signalrcore#6]
The example code and the `signalrcore` library itself don't work in Python 2.7.

*First issue*:
The example code contains a callback function `print`. This is not supported in Python 2.7. However this is easy to solve by just changing this into a self made callback function.

*Second issue*:
More important, the `signalrcore` library cannot be imported in a Python 2.7 context. So for example the first statement in the example code:

```
from signalrcore.hub_connection_builder import HubConnectionBuilder
```

will result in this error:
```
Traceback (most recent call last):
  File "websocket_example.py", line 1, in <module>
    from signalrcore.hub_connection_builder import HubConnectionBuilder
  File "C:\Python27\lib\site-packages\signalrcore\hub_connection_builder.py", line 4, in <module>
    from .hub.auth_hub_connection import AuthHubConnection
  File "C:\Python27\lib\site-packages\signalrcore\hub\auth_hub_connection.py", line 4, in <module>
    from .errors import UnAuthorizedHubError, HubError
  File "C:\Python27\lib\site-packages\signalrcore\hub\errors.py", line 1, in <module>
    class HubError(ConnectionError):
NameError: name 'ConnectionError' is not defined
```

`ConnectionError` is a standard exception in Python3, but **not** in Python2.

**To Reproduce**
1) make sure both Python v2.7 and Python v3.7 versions are installed on your system
2) make sure the `signalrcore` client library is installed for both versions
3) run example code in Python3 context:

```
C:\project\websocket>py -3 websocket_example.py
Enter your server url(default: ws://localhost:62342/chathub):
```

This is OK

4) run example code in Python2 context:

```
C:\project\websocket>py -2 websocket_example.py
Traceback (most recent call last):
  File "websocket_example.py", line 1, in <module>
    from signalrcore.hub_connection_builder import HubConnectionBuilder
  File "C:\Python27\lib\site-packages\signalrcore\hub_connection_builder.py", line 4, in <module>
    from .hub.auth_hub_connection import AuthHubConnection
  File "C:\Python27\lib\site-packages\signalrcore\hub\auth_hub_connection.py", line 4, in <module>
    from .errors import UnAuthorizedHubError, HubError
  File "C:\Python27\lib\site-packages\signalrcore\hub\errors.py", line 1, in <module>
    class HubError(ConnectionError):
NameError: name 'ConnectionError' is not defined
```

This is clearly NOK

**Expected behavior**
Library should work in both Python2 and Python3 context as also stated by the pypi website: https://pypi.org/project/signalrcore/

**Desktop:**
 - OS: Windows 7 Pro

**Comment by mandrewcito** [comment]
Solved on the version 0.7.0, was an built in exception from py3

**Comment by gvanderpol** [comment]
Are you sure this is solved? I used your latest version 0.7.0 and if I try to import the HubConnectionBuilder class as stated in the example, I still get the undefined 'ConnectionError'.

pip list (omitted full output):
```
C:\project\websocket>py -2 -m pip list
..
..
Package                       Version
----------------------------- ---------
..
..
signalrcore                   0.7.0
..
..
``` 

REPL (python2):
```
C:\project\websocket>py -2
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>>
>>> from signalrcore.hub_connection_builder import HubConnectionBuilder
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\signalrcore\hub_connection_builder.py", line 4, in <module>
    from .hub.auth_hub_connection import AuthHubConnection
  File "C:\Python27\lib\site-packages\signalrcore\hub\auth_hub_connection.py", line 4, in <module>
    from .errors import UnAuthorizedHubError, HubError
  File "C:\Python27\lib\site-packages\signalrcore\hub\errors.py", line 1, in <module>
    class HubError(ConnectionError):
NameError: name 'ConnectionError' is not defined
>>>
```

And it is not only this undefined exception. I also see that some internal modules uses other stuff from Python3, which are not available in Python2, like in the `signalrcore.helpers` module this import:

```
from urllib import parse
```

Will lead to this error in Python2:
```
C:\Python27>py -2
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> from urllib import parse
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name parse
>>>
```

This `parse` function is in Python2 available in `urlparse`, so for Python2 this statement should be:
```
C:\Python27>py -2
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> import urlparse as parse
>>>
```

So, there might be more stuff in the signalrcore, which is not Python2 compatible.

**Comment by mandrewcito** [comment]
This was fixed on previous version (since 0.7.4).   Thanks for reporting it !

**Comment by saulkali** [comment]
hi , mandrewcito, you know why HubConnectionBuilder, create new event on, when this call a one event, for example 

i create my client, this show in the log this.

 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
SignalRCoreClient - DEBUG - {"type": 6}

when the server call the event and my client receive message, this create new event on, you now what is happening?

in my log show this when receive message. 


 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
SignalRCoreClient - DEBUG - {"type": 6}
SignalRCoreClient - DEBUG - {"type": 6}


if you send message, the client duplicate connection


 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
SignalRCoreClient - DEBUG - {"type": 6}
SignalRCoreClient - DEBUG - {"type": 6}
SignalRCoreClient - DEBUG - {"type": 6}

sorry for my english 


lo dejo tambien en spanish, tengo el problema de cuando creo una conexion con este modulo todo inicia bien, con el siguiente log.

 - SignalRCoreClient - DEBUG - Sending message <signalrcore.messages.ping_message.PingMessage object at 0x000001D49942DE50>
SignalRCoreClient - DEBUG - {"type": 6}


entonces cuando el servidor me manda un mensaje a mi cliente y este ejecuta el evento on, todo llega de manera correcta solo que cuando llega duplica la conexion y me crea nuevas lo cual se duplica si tenia solo 1 este pasa a crear 2 de 2 pasa a 4 de 4 a 8, con solo 4 o 6 peticiones se empieza a romper el codigo. por tantos eventos on que se duplican 


dejo mi codigo 

print("connect websocket")
            self.hub_connection = HubConnectionBuilder() \
                .with_url(ApiSettings.host_base + ApiSettings.hub_signal_r,options={
                        "verify_ssl":False
                }) \
                .configure_logging(logging.DEBUG) \
                .with_automatic_reconnect({
                "type": "raw",
                "keep_alive_interval": 10,
                "reconnect_interval": 5,
                "max_attempts": 5
            }).build()

            self.hub_connection.on_open(lambda: print("connection opened and handshake received ready to send messages"))
            self.hub_connection.on_close(lambda: print("connection closed"))
            self.hub_connection.on_error(lambda data: print(f"An exception was thrown closed{data.error}"))

            self.hub_connection.on(ApiSettings.call_turn_endpoint + self.id_room.__str__(), callback_function=  self.on_call_turn)
            self.hub_connection.on(ApiSettings.get_turn_endpoint + self.id_room.__str__(), callback_function=  self.on_get_turn)
            self.hub_connection.start()
            print("signal connected :)")

please can you helpe me? what is happining
